### PR TITLE
try to properly handle the case where the bits are not found

### PR DESCRIPTION
**Pull request by peter-meemo**

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -14,8 +18,19 @@ var (
 	ErrServer             = errors.New("server error")
 	ErrServiceUnavailable = errors.New("service unavailable")
 	ErrUnknown            = errors.New("unknown error")
+	ErrBitStateNotFound = errors.New("bit state not found")
```
Comment (robsonmeemo) [inline error.go:21]: Wrong identation

File: query.go [hunk 2]
```diff
@@ -70,8 +71,16 @@ func (client *Client) QueryTwoBits(deviceToken string, result *QueryTwoBitsResul
...
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	err = newErrorForQuery(resp.StatusCode, string(respBody))
+	if err == ErrBitStateNotFound {
+		return err
+	}
+	if err != nil {
+		return fmt.Errorf("devicecheck: %w", err)
```
Comment (robsonmeemo) [inline query.go:83]: Shouldn't you just return err here?

File: error.go [hunk 3]
```diff
@@ -14,8 +18,19 @@ var (
...
 )
 
+func newErrorForQuery(code int, body string) error {
+	if code != http.StatusOK {
+		return newError(code)
+	}
+	if body == bitStateNotFoundStr {
```
Comment (robsonmeemo) [inline error.go:28]: This error is returned with a 200 status?

File: query.go (hunk 2 above)
Comment (peter-meemo) [reply query.go:83]: I would if I were writing from scratch, but this is trying to be a minimal change to the library. Do you think I should?

File: error.go (hunk 3 above)
Comment (peter-meemo) [reply error.go:28]: Yes. See 'Understand the HTTP Response Codes' here:
https://developer.apple.com/documentation/devicecheck/accessing_and_modifying_per-device_data

It's not technically an error from the API perspective, we could handle it by populating a bit in the response object instead, but I do think this is a cleaner way.

File: error.go (hunk 1 above)
Comment (peter-meemo) [reply error.go]: done

**Review by robsonmeemo — APPROVED** [review]

**Pull request merged**